echo/README.md

```diff
@@ -1,5 +1,233 @@
+    Can Be Used                                               immediately 
+    Version                                                     0.2.2 (0) 
+    Call Alias                                                  (Not set) 
+    ```
+
+    We are after the module ID: `MCUDTMOOZCVAM5EBNN4U3X2OGHNIY3BKPEW66HY4RTCYYVWXOE7ESVDQ`
+
+7. Swap the remote actor with our newly created one in the manifest file and don't forget to change the actor ID:
+
+    ```yaml
+    actors:
+        - "./target/wasm32-unknown-unknown/release/echo_s.wasm"
+    # ...
+    actor: ${ECHO_ACTOR:MCUDTMOOZCVAM5EBNN4U3X2OGHNIY3BKPEW66HY4RTCYYVWXOE7ESVDQ}
+    # ...
+    ```
+
+8. Now it's time to run our actor:
+
+    ```bash
+    $ wasmcloud -m manifest.yaml
+    ```
+
+9. As a final step we will send another request to `http://localhost:8080`:
+
+    ```bash
+    curl http://localhost:8080/
+    ```
+
+## Using `wash` for better developer experience
+
+Previously we were using a declarative deployment model that did not allow us to modify resources at runtime. In this section, we will use `NATS` to allow `wasmcloud` to utilize self-managing [lattice network](https://wasmcloud.dev/reference/lattice/), and we can use `wash` to interact with the `wasmcloud` host via the lattice control interface (`ctl`).
+
+We will also use a local docker registry to store our echo actor to illustrate interacting with a local development version of an OCI registry.
+
+1. Since we are going to use `docker-compose` to run NATS server and  image registry, you need to have `wasmcloud`, `wash`, `docker` and `docker-compose` installed on your machine.
+
+2. You can either use `docker-compose.yml` file in the project's directory or create one with the following content:
+
+    ```yaml
+    version: "3"
+    services:
+      registry:
+        image: registry:2
+        ports:
+        - "5000:5000"
+      nats:
+        image: nats:2.1.9
+        ports:
+          - "6222:6222"
+          - "4222:4222"
+          - "8222:8222"
+    ```
+
+    We will run our containers in detached mode to reduce clutter.
+
+    ```bash
+    $ docker-compose up -d
+    ```
+
+    Start a REPL session using `wash up` command:
+
+    ```bash
+    $ wash up
+    ```
+
+    Check if session is running:
+
+    ```bash
+    $ ctl get hosts
+    ```
+
+    It will output something like:
+
+    ```bash
+    Host ID                                                   Uptime (seconds)
+    NDGPVEOZKAYT6JRL2C4BDVSW3PBBOYTWRLC2QFCSLXWDX4KCIRHM6WCZ  1357
+    ```
+
+3. Please open a new terminal, as the REPL is running in the current one. We will build our actor first, then will sign it:
+
+    ```bash
+    $ cargo build --release
+    $ wash claims sign ./target/wasm32-unknown-unknown/release/echo.wasm -c wasmcloud:httpserver --name "echo" --ver 0.2.2
+    ```
+
+    Or you can use `make` file in project's directory for convenience:
+
+    ```bash
+    make release VERSION=0.2.2
```

Comment:
I love this. This is the first time I've seen the use of the `VERSION` variable used anywhere in our docs.